password_window/PasswordWindow.py

```diff
@@ -0,0 +1,77 @@
+from tkinter import Toplevel
+from external.constants import DEFAULT_BACKGROUND_COLOR, DEFAULT_WINDOW_GEOMETRY_SIZE
+from title_label.TitleLabel import TitleLabel
+from custom_entry.CustomEntry import CustomEntry
+from custom_button.CustomButton import CustomButton
+from messagee_box.PasswordMessageBox import PasswordMessageBox
+import json
+
+
+class PasswordWindow(Toplevel):
+    """Class, which represents password window. It inherits from Toplevel object"""
+
+    def __init__(self, main_app):
+        """Construct new PasswordWindow object
+
+        Params:
+            main_app (App): Main application object"""
+        Toplevel.__init__(self, background=DEFAULT_BACKGROUND_COLOR)
+
+        self.__main_app = main_app
+        self.__main_app.get_root_window().withdraw()
+
+        self.__title_label = TitleLabel(self)
+        self.__title_label.move(x_coordinate=65, y_coordinate=45)
+        self.__title_label.set_text("Login window")
+
+        self.__username_entry = CustomEntry(self, x_coordinate=40, y_coordinate=150)
+        self.__username_entry.set_text("Username: ")
+
+        self.__password_entry = CustomEntry(self, x_coordinate=40, y_coordinate=200)
+        self.__password_entry.set_text("Password: ")
+        self.__password_entry.set_mode(CustomEntry.Mode.PASSWORD)
+
+        self.__login_button = CustomButton(
+            self,
+            "Login",
+            self.__login,
+            DEFAULT_WINDOW_GEOMETRY_SIZE,
+            DEFAULT_WINDOW_GEOMETRY_SIZE,
+        )
+
+        self.__exit_button = CustomButton(
+            self,
+            "Exit",
+            lambda: exit(),
+            DEFAULT_WINDOW_GEOMETRY_SIZE,
+            DEFAULT_WINDOW_GEOMETRY_SIZE,
+        )
+
+        self.__password_message_box = PasswordMessageBox(
+            "Oops!", "Wrong password! Try again!"
+        )
+
+        data_file = open(
+            "data/login_data.json"
+        )  # todo: this path doesn't work properly with desktop shortcut - fix it immediately
```

Comment:
how about sth like: path_to_file = str(getcwd()+'/data/login_data.json')?

Reply:
Sorry but it has been my mistake and it doesnt work, try sth like:
from pathlib import Path
data_file = open(Path(__file__).parent.parent/'data/login_data.json')
previous option didn't work because of problem with importing, then the file main state is on the file on which  we've currently been so 'getcwd' returns us path to desktop (place where is our shortcut) not to the main file.

Reply:
Other possibility :
import __ main __ (our main file is main.py)
data_file = open(Path(__main__.__ file __).parent/'data/login_data.json')